### PR TITLE
Request to merge Validations

### PR DESCRIPTION
**Pull request by Kingstalux**
- Added validations to post and user models
- Wrote tests for validations and methods of models
- Added flash messages
- Fix n+1 problem for the post index page.

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
     <%= render "partials/user", user: @user %>
 
-    <% @posts.each do |post| %>
+    <% @posts.includes(:comments).each do |post| %>
```
Comment (elisha2kyakpo1) [inline app/views/posts/index.html.erb:17]: - [x] [OPTIONAL] We can avoid having logic in the views by taking this piece of code in the index action of the post controller.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi  @Kingstalux,

Great job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

- The PR title and description looks great 👍
- The model validations are well written ✔️
- The n + 1 query issue has been addressed 👍

## Required Changes ♻️
_Check the comments under the review._

- [x] Let's look at this project instruction once more `Add unit specs for all of your models' methods and validations.` I did not see the model method tests.
- [x]  Let's implement this project requirement `Add flash messages in the create actions in all your controllers.`

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kingstalux** [comment]
- Added tests for model methods.
- The flash messages are already present in create actions of my controllers, so I did not have to add them again. You can check those in the controller file in the repo.

**Review by bimbolabuari — APPROVED** [review]
# Project Approved 🚀

Hi @Kingstalux🥇,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**